native/src/main/java/io/ballerina/stdlib/sql/datasource/SQLDatasource.java

```diff
@@ -195,6 +195,12 @@ public static Connection getConnection(boolean isInTrx, TransactionResourceManag
                 conn = ((SQLTransactionContext) txContext).getConnection();
             }
         } catch (SQLException e) {
+            if (e.getCause() != null &&
+                e.getCause().getMessage().contains("Data source rejected establishment of connection")) {
+                throw new SQLException("error while getting the connection for " + Constants.CONNECTOR_NAME +
```

Comment:
Unable to test using HSQLDB, as apparently it does not allow for setting a global connection limit.

Reply:
Is this an error message given by HikkariCP or specific DB? If it is specific DB we have to handle this in specific DB connectors

Reply:
It is an error message given by HikkariCP
https://github.com/brettwooldridge/HikariCP/blob/215ce945c6fb2116aa288ddf9b790675347d0047/src/main/java/com/zaxxer/hikari/pool/HikariPool.java#L154

Reply:
Ack

Reply:
Amended PR in [b77ac24](https://github.com/ballerina-platform/module-ballerina-sql/pull/584/commits/b77ac24019c3f14dea0cb108d56eb143c459e2df) to return the error message generated by the driver.

Reply:
Shall we give samples in the original issue and say why we couldn't proceed with this improvement